// @coinbase/onchainkit/dist/lib/core/getFrameAccountAddress.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const getFrameAccountAddress_1 = require("./getFrameAccountAddress");
const mock_1 = require("./mock");
const neynarUserFunctions_1 = require("../utils/neynar/user/neynarUserFunctions");
jest.mock('../utils/neynar/user/neynarUserFunctions', () => {
    return {
        neynarBulkUserLookup: jest.fn(),
    };
});
jest.mock('@farcaster/hub-nodejs', () => {
    return {
        getSSLHubRpcClient: jest.fn().mockReturnValue({
            validateMessage: jest.fn(),
        }),
        Message: {
            decode: jest.fn(),
        },
    };
});
describe('getFrameAccountAddress', () => {
    const fakeFrameData = {
        trustedData: {},
    };
    const fakeApiKey = {
        NEYNAR_API_KEY: '1234',
    };
    it('should return the first verification for valid input', async () => {
        const fid = 1234;
        const addresses = ['0xaddr1'];
        (0, mock_1.mockNeynarResponse)(fid, addresses, neynarUserFunctions_1.neynarBulkUserLookup);
        const response = await (0, getFrameAccountAddress_1.getFrameAccountAddress)(fakeFrameData, fakeApiKey);
        expect(response).toEqual(addresses[0]);
    });
    it('when the call from farcaster fails we should return undefined', async () => {
        const fid = 1234;
        const addresses = ['0xaddr1'];
        const { validateMock } = (0, mock_1.mockNeynarResponse)(fid, addresses, neynarUserFunctions_1.neynarBulkUserLookup);
        validateMock.mockClear();
        validateMock.mockResolvedValue({
            isOk: () => {
                return false;
            },
        });
        const response = await (0, getFrameAccountAddress_1.getFrameAccountAddress)(fakeFrameData, fakeApiKey);
        expect(response).toEqual(undefined);
    });
});
